File: send_audio.js
const fs = require('fs');
const WebSocket = require('ws');

// Read base64 audio from file
const base64Audio = fs.readFileSync('audio.b64', 'utf8').replace(/\r?\n/g, '');

// Connect to the local WebSocket endpoint
const ws = new WebSocket('ws://localhost:62564/api/call/stream');

ws.on('open', function open() {
  // Send start event
  ws.send(JSON.stringify({ event: 'start', start: { callSid: 'testcall' } }));

  // Send media event with base64 audio payload
  ws.send(JSON.stringify({
    event: 'media',
    start: { callSid: 'testcall' },
    media: { payload: base64Audio }
  }));

  // Optionally, send stop event
  ws.send(JSON.stringify({ event: 'stop', stop: { callSid: 'testcall' } }));
});

ws.on('message', function message(data) {
  console.log('Received:', data.toString());
});

ws.on('close', function close() {
  console.log('WebSocket closed.');
});

ws.on('error', function error(err) {
  console.error('WebSocket error:', err);
});

File: send_audio.py
import websocket
import json

# Read base64 audio from file
with open('audio.b64', 'r') as f:
    base64_audio = f.read().replace('\n', '').replace('\r', '')

# Connect to the local WebSocket endpoint
ws = websocket.create_connection('ws://localhost:62564/api/call/stream')

# Send start event
ws.send(json.dumps({"event": "start", "start": {"callSid": "testcall"}}))

# Send media event with base64 audio payload
ws.send(json.dumps({
    "event": "media",
    "start": {"callSid": "testcall"},
    "media": {"payload": base64_audio}
}))

# Optionally, send stop event
ws.send(json.dumps({"event": "stop", "stop": {"callSid": "testcall"}}))

# Print any response from the server
try:
    while True:
        msg = ws.recv()
        print("Received:", msg)
except websocket._exceptions.WebSocketConnectionClosedException:
    print("WebSocket closed.")

ws.close()
